src/parser/expression.js

```diff
@@ -776,6 +799,11 @@ export default class ExpressionParser extends LValParser {
     node.callee = this.parseNoCallExpr();
     const optional = this.eat(tt.questionDot);
 
+    if (this.isRelational("<")) {
+      const typeArguments = this.parseTypeArguments();
```

Comment:
What if "<" is just a relational operator? Like in this case:
```js
function A() { }
A.prototype.valueOf = () => 2;
var T = 3;

new A < T;
```
I see that in `typescript.js#parseSubscript` you considered that edge case when parsing call expressions.
(I have not tested if this actually works or not, but I got confused by this difference in the two methods)

---

Anyway, you can also extract these lines (799-818 or 802-812) to a separate method and overwrite it in `typescript.js`, thus removing the type arguments check from `expression.js`